samples/net/http_server/src/main.c

```diff
@@ -167,6 +171,31 @@ static int http_response_soft_404(struct http_server_ctx *ctx)
 			     HTML_FOOTER);
 }
 
+static int http_response_auth(struct http_server_ctx *ctx)
+{
+	return http_response(ctx, HTTP_STATUS_200_OK, HTML_HEADER
+			     "<h2><center>user: "HTTP_AUTH_USERNAME
+			     ", password: "HTTP_AUTH_PASSWORD
+			     "</center></h2>" HTML_FOOTER);
+}
+
+int http_response_401(struct http_server_ctx *ctx)
+{
+	/* Wait 2 secs for the reply with proper credentials */
+	return http_response_wait(ctx, HTTP_401_STATUS_US, NULL, K_SECONDS(2));
+}
+
+static int http_basic_auth(struct http_server_ctx *ctx)
+{
+	const char *auth_str = "Authorization: Basic "HTTP_AUTH_CREDENTIALS;
+
+	if (strstr(ctx->req.field_values[0].key, auth_str)) {
```

Comment:
One thing that prevents the use of memcmp() or strncmp() here, is the way field_values array is constructed. It actually contains pointers to header values and those values are not null terminated (so original read data header data is not modified by the library). This means that we can just use the strstr() which will go through all the individual headers in one go. We might want to change this if needed, but later.